python/ray/data/datasource/datasource.py

```diff
@@ -158,19 +184,42 @@ def make_block(start: int, count: int) -> Block:
                 schema = int
             else:
                 raise ValueError("Unsupported block type", block_format)
+            meta = BlockMetadata(
+                num_rows=count,
+                size_bytes=8 * count,
+                schema=schema,
+                input_files=None)
             read_tasks.append(
                 ReadTask(
-                    lambda i=i, count=count: make_block(i, count),
-                    BlockMetadata(
-                        num_rows=count,
-                        size_bytes=8 * count,
-                        schema=schema,
-                        input_files=None)))
+                    lambda i=i, count=count, meta=meta: [(
+                        ray.put(make_block(i, count), _owner=owner), meta)],
+                    meta))
             i += block_size
 
         return read_tasks
 
 
+@ray.remote(num_cpus=0, placement_group=None)
+class _DesignatedBlockOwner:
+    def ping(self):
+        return "ok"
+
+
+# We manually assign the owner so ray.put() blocks do not fate share with the
+# worker that created them.
```

Comment:
I think we're talking about different things here. I'm asking whether assigning an actor as the block owner for `ray.put()` is needed for correctness over doing a normal `ray.put()` (no owner assignment). In the latter case, the worker that executes the (read/map) task will be considered the owner, and that worker will be kept alive as long as `ref` is in scope, right?

Reply:
The only two reasons I can think of for doing ownership assignment to a detached actor are:
1. An easier path to fault-tolerance than having to fully support owner death for borrowed refs, which I think is otherwise required.
2. An optimization to prevent the cluster from filling up with a lot of otherwise-idle workers that own these put refs.

Reply:
> I think we're talking about different things here. I'm asking whether assigning an actor as the block owner for ray.put() is needed for correctness over doing a normal ray.put() (no owner assignment). 

Yes it is needed for correctness. Otherwise, the block will fate share with workers/actors, which is undesirable since it introduces multiple points of failure, and also actors get killed as soon as the map stage finishes.

Reply:
For my own understanding: the multiple points of failure makes sense to me, but isn't the worker that underlies an actor kept alive if that worker owns objects that are still in scope, even if the actor is terminated? I had always assumed that was the case but I've never checked.

Reply:
Ah so we don't wait for refs to go out of scope for actors, since the actor heap can hold on to refs for forever. Got it.